src/fragments/lib/auth/js/advanced.mdx

```diff
@@ -4,53 +4,145 @@ You can take specific actions when users sign-in or sign-out by subscribing auth
 
 ## Identity Pool Federation
 
-You can alternatively use `Auth.federatedSignIn()` to get AWS credentials directly from Cognito Federated Identities and not use User Pool federation. If you have logged in with `Auth.signIn()` you **can not** call  `Auth.federatedSignIn()` as Amplify will perform this federation automatically for you in the background.
+You can alternatively create your own custom credentials provider to get AWS credentials directly from Cognito Federated Identities and not use User Pool federation. You must supply the custom credentials provider to Amplify via the `Amplify.configure` method call. Below, you can see sample code of how such a custom provider can be built to achieve the use case. 
 
-In general, if you are using Cognito User Pools to manage user Sign-Up and Sign-In, you should only call `Auth.federatedSignIn()` when using OAuth flows or the Hosted UI.
 
 ```js
-import { Auth } from 'aws-amplify';
-
-// To derive necessary data from the provider
-const {
-  token, // the token you get from the provider
-  domainOrProviderName, // Either the domain of the provider(e.g. accounts.your-openid-provider.com) or the provider name, for now the library only supports 'google', 'facebook', 'amazon', 'developer'
-  expiresIn, // the time in ms which describes how long the token could live
-  user,  // the user object you defined, e.g. { username, email, phone_number }
-  identity_id // Optional, the identity id specified by the provider
-} = getFromProvider(); // arbitrary function
-
-async function getCognitoCredentials() {
-  try {
-    const cred = await Auth.federatedSignIn(
-      domainOrProviderName,
-      {
-        token,
-        identity_id, // Optional
-        expires_at: expiresIn * 1000 + new Date().getTime() // the expiration timestamp
-      },
-      user
-    );
-    console.log(cred);
-
-    const authenticatedUser = await Auth.currentAuthenticatedUser();
-
-    console.log(authenticatedUser);
-  } catch(err) {
-    console.log(err);
+import { Amplify } from "aws-amplify";
+import {
+  fetchAuthSession,
+  AWSCredentialsAndIdentityIdProvider,
+  AWSCredentialsAndIdentityId,
+  TokenProvider,
+  AuthTokens,
+} from "aws-amplify/auth";
+
+import { CognitoIdentity } from "@aws-sdk/client-cognito-identity";
```

Comment:
I would add a comment that customers need to install `@aws-sdk/client-cognito-identity` package